packages/mds-compliance-service/service/provider.ts

```diff
@@ -0,0 +1,124 @@
+import logger from '@mds-core/mds-logger'
+import { ServiceResult, ServiceException, ServiceProvider, ProcessController } from '@mds-core/mds-service-helpers'
+import { UUID } from '@mds-core/mds-types'
+import { isDefined } from '@mds-core/mds-utils'
+import { providerName } from '@mds-core/mds-providers'
+import {
+  ComplianceService,
+  GetComplianceSnapshotsByTimeIntervalOptions,
+  GetComplianceViolationPeriodsOptions,
+  ComplianceAggregateDomainModel,
+  ComplianceViolationPeriodDomainModel
+} from '../@types'
+import { ComplianceRepository } from '../repository'
+import {
+  ValidateComplianceSnapshotDomainModel,
+  ValidateGetComplianceSnapshotsByTimeIntervalOptions
+} from './validators'
+import { ComplianceViolationPeriodEntityToDomainCreate } from '../repository/mappers'
+
+interface ComplianceAggregateMap {
+  [k: string]: ComplianceViolationPeriodDomainModel[]
+}
```

Comment:
Could you just be using `Map<string, ComplianceViolationPeriodDomainModel[]>`

Reply:
Didn't know about that type, thanks

Reply:
`Map<string, ComplianceBlahBlah[]>` doesn't seem to support `{}` as a possible value.

Reply:
It’s a Javascript Map. You initialize it using `new Map<string, ComplianceBlahBlah[]>()`.

Reply:
Ah, I thought it was a TS thing like `Omit`.